### PR TITLE
Add support for sysctl.

### LINKED ISSUE #81: Add support for sysctl
https://www.nomadproject.io/docs/drivers/docker#sysctl

### PR DESCRIPTION
**Pull request by shishir-a412ed**
Fixes #81

### CODE REVIEW

**Comment by chuckyz** [comment]
Why are some packages pulled in twice?

File: containerd/driver.go [hunk 1]
```diff
@@ -151,23 +153,24 @@ type Mount struct {
...
+	Image          string             `codec:"image"`
+	Command        string             `codec:"command"`
+	Args           []string           `codec:"args"`
+	CapAdd         []string           `codec:"cap_add"`
+	CapDrop        []string           `codec:"cap_drop"`
+	Cwd            string             `codec:"cwd"`
+	Devices        []string           `codec:"devices"`
+	Seccomp        bool               `codec:"seccomp"`
+	SeccompProfile string             `codec:"seccomp_profile"`
+	Sysctl         hclutils.MapStrStr `codec:"sysctl"`
```
Comment (chuckyz) [inline containerd/driver.go:165]: Why is this a `MapStrStr` while the above hclspec has `list(map(string))` ?

**Comment by shishir-a412ed** [comment]
@chuckyz 
> Why are some packages pulled in twice?

Which ones?

File: containerd/utils.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+func WithSysctls(sysctls map[string]string) oci.SpecOpts {
+	return func(ctx context.Context, client oci.Client, c *containers.Container, s *specs.Spec) error {
+		if s.Linux == nil {
+			s.Linux = &specs.Linux{}
+		}
+		if s.Linux.Sysctl == nil {
+			s.Linux.Sysctl = make(map[string]string)
+		}
+		for k, v := range sysctls {
+			s.Linux.Sysctl[k] = v
```
Comment (chuckyz) [inline containerd/utils.go:58]: Are there any cases of nested sysctls, or is the end result always a single value or string?

File: containerd/driver.go (hunk 1 above)
Comment (shishir-a412ed) [reply containerd/driver.go:165]: This is how it is in `docker driver`

https://github.com/hashicorp/nomad/blob/main/drivers/docker/config.go#L396 and https://github.com/hashicorp/nomad/blob/main/drivers/docker/config.go#L465

This works pretty well, as the input (`sysctl`) is directly populated into a `go` map, which I can use to set the `map` in the container spec.

File: containerd/utils.go (hunk 2 above)
Comment (shishir-a412ed) [reply containerd/utils.go:58]: The end result is a `map` of `key` and `values`. We get an input map (`sysctls`) to the function, and we iterate over the input map and set the container spec map `s.Linux.Sysctl`.

**Comment by chuckyz** [comment]
@shishir-a412ed just perusing the diff, I will do research on my own around go pkg as I feel it's juts an element of that. :)

**Review by chuckyz — APPROVED** [review]

**Pull request merged**